monitoring/nethealth.go

```diff
@@ -303,7 +303,6 @@ func (c *nethealthChecker) fetchNethealthMetrics(ctx context.Context) (res []byt
 				return net.Dial("unix", c.NethealthSocketPath)
```

Comment:
I think this is inferior to using the net.Dialer directly (and thus making good use of the context):
```go
DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
	var d net.Dialer
	return d.DialContext(ctx, "unix", c.NethealthSocketPath)
}
```